### ISSUE TITLE
pandas-gbq handles nulls in numeric columns differently from pandas

### ISSUE THREAD

**Issue by tswast** [pydata/pandas-gbq#174]
[Pandas encodes missing / null data as NaN in numeric columns](https://pandas.pydata.org/pandas-docs/stable/missing_data.html).

pandas-gbq expects the type of a column containing Nulls to be object.
https://github.com/pydata/pandas-gbq/blob/f301442082bab62c793b6a80cf00c03f97938609/tests/system.py#L295-L302

Shouldn't pandas-gbq align with the choice of pandas in this case?

**Comment by tswast** [comment]
Note that for datetime/timestamp columns the pandas-gbq behavior is a aligned with [pandas by using NaT for missing dates](https://pandas.pydata.org/pandas-docs/stable/missing_data.html#datetimes).

https://github.com/pydata/pandas-gbq/blob/f301442082bab62c793b6a80cf00c03f97938609/tests/system.py#L383-L387

**Comment by maxim-lian** [comment]
> Shouldn't pandas-gbq align with the choice of pandas in this case?

You do then have all ints as floats. If you want floats with `NaN`s, you can cast to float in the query. And with the change there's no such escape hatch.

But I'm +0.2 to make the change - it's far more likely that someone would want a float column than an object column.

**Comment by tswast** [comment]
> You do then have all ints as floats.

Yeah, I find that a bit odd, but it's what the DataFrame constructor does in the case of missing values.

```
In [1]: import pandas as pd

In [2]: df_int = pd.DataFrame([[1], [2], [3]])

In [3]: df_int.dtypes
Out[3]:
0    int64
dtype: object

In [4]: df_int_null = pd.DataFrame([[1], [None], [3]])

In [5]: df_int_null.dtypes
Out[5]:
0    float64
dtype: object
```

**Comment by max-sixty** [comment]
Closed by #224

**Comment by msegado** [comment]
Quick FYI, it looks like the docs still describe the old behavior:
https://pandas-gbq.readthedocs.io/en/stable/intro.html#integer-and-boolean-na-handling
